### PR TITLE
Geany: 1.30 -> 1.30.1

### PR DESCRIPTION
**Pull request by frlan**
###### Motivation for this change

This PR updates Geany to 1.30 as there was a bug found in 1.30 causing e.g. calltips popping up on wrong display when using Geany in a multi-display environment and not running on left top monitor.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by ndowens — APPROVED** [review]

File: pkgs/applications/editors/geany/default.nix [hunk 1]
```diff
@@ -13,8 +15,9 @@ stdenv.mkDerivation rec {
   };
 
   NIX_LDFLAGS = if stdenv.isDarwin then "-lintl" else null;
-
-  buildInputs = [ gtk2 which pkgconfig intltool file ];
+  
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ gtk2 which intltool file ];
```
Comment (Mic92) [inline pkgs/applications/editors/geany/default.nix]: intltool also belongs to nativeBuildInputs.

Comment (ndowens) [inline pkgs/applications/editors/geany/default.nix]: @Mic92 was thinking it may be, will fix it :) running nox-review right now before I merge

**Comment by ndowens** [comment]
@frlan nox-review gives me: checking pkg-config is at least version 0.9.0... yes
configure: checking for guile 2.0
configure: checking for guile 1.8
configure: error:
No Guile development packages were found.

**Review by ndowens — DISMISSED** [review]
Seems to be missing deps

**Pull request merged**